### PR TITLE
Feature/swiftui support

### PR DESCRIPTION
**Pull request by ZvonimirMedak**
## Summary

This PR adds SwiftUI support to the Sentinel library, with the refactor of the currently available tools to SwiftUI.
There are a lot of files changed, but some of them are minor changes, and some of them have been deleted

**Related issue**:  None

### Type

- [-] **Feature**: This pull request introduces a new feature.
- [-] **Bug fix**: This pull request fixes a bug.
- [x] **Refactor**: This pull request refactors existing code.
- [-] **Documentation**: This pull request updates documentation.
- [-] **Other**: This pull request makes other changes.

#### Additional information

- [X] This pull request introduces a **breaking change**.

### Description

This PR contains three major updates to the current state of the library:
1. The Tool protocol now contains a ViewBuilder for the SwiftUI view representing the Tool. Removing the UIKit coupling with the presentPreview
2. The ToolTableItem is an Enum now which contains four primary cases (Navigation, Toggle, CustomInfo, and Performance) which can be added to the List view to be easily shown in Sentinel. Also, includes a new `.custom` case which adds support for a custom View, and behavior
3. The currently available tools have been refactored to use SwiftUI. Although, EmailSender, and Custom location have been wrapped in a UIViewRepresentable since there's no support for MessagesUI in SwiftUI, and CustomLocation would lose some of its functionalities due to lack of MapReader on iOS 14

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [x] I have added necessary tests for the changes introduced (if applicable).
- [-] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

The code documentation will be added in a separate PR

### CODE REVIEW

File: Sentinel/Classes/TextEditing/TextEditingTool.swift [hunk 1]
```diff
@@ -5,19 +5,23 @@
 //  Created by Vlaho Poluta on 31/07/2020.
 //
 
-import UIKit
+import SwiftUI
 
 @objcMembers
-public class TextEditingTool: NSObject, Tool {
-    
+public final class TextEditingTool: NSObject, Tool {
```
Comment (nikolamajcen) [inline Sentinel/Classes/TextEditing/TextEditingTool.swift:11]: There is no need for subclassing the `NSObject` class anymore - this can be defined as a struct.

_Note: check other usages of `NSObject` and update if necessary._

**Review by nikolamajcen — CHANGES_REQUESTED** [review]
Hey @ZvonimirMedak, amazing job! You did a heavy-lifting regarding the project update and its support for SwiftUI.

Still, I've added some comments regarding the implementation - please check them out.

_Note: I was unable to build the project due to multiple files (`NavigationToolTableItem` duplication and location issues)._

File: Sentinel/Classes/Core/CustomInfoTool.swift [hunk 2]
```diff
@@ -24,25 +29,24 @@ public class CustomInfoTool: NSObject, Tool {
...
-    
-    // MARK: - Public properties
-    
-    public func presentPreview(from viewController: UIViewController) {
-        let toolTable = createToolTable(with: info)
-        toolTable.presentPreview(from: viewController)
-    }
+}
+
+// MARK: - Extensions -
```
[the remark this answers is not in the document]
Comment (ZvonimirMedak) [reply Sentinel/Classes/Core/CustomInfoTool.swift:34]: I never noticed this, thank you 🙇

**Comment by ZvonimirMedak** [comment]
Hi @nikolamajcen,
I've fixed the comments which weren't fixed in the #44 PR. The objc comments, and the warnings are fixed in the next PR. I'm not sure how I didn't catch the models which weren't properly added to the package. Hopefully, now you can test it out properly 😄 

I left a question regarding the Extension MARKs, and we can sync later on regarding the fonts. If you want me to merge the #44 before you review this one again just let me know 😄

**Comment by nikolamajcen** [comment]
Hey @ZvonimirMedak, I'm fine with merging #44 before this one so that I can check the final implementation with removal and updates for the mentioned parts of the code (e.g., obj-c support, etc.).

Regarding the unused `MARK`s - you can remove those that are empty (e.g. extension marks in multiple cases).

File: Sentinel/Classes/Core/Internal/Image+Assets.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
+//
+//  Image+Assets.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 10.12.2024..
+//
+
+import SwiftUI
+
+// There's an issue with the generated assets, and local cocoapods which should be resolved in Xcode 16.1
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/Internal/Image+Assets.swift:10]: Open issue for this, so we can track this (Productive or Issues here).

File: Sentinel/Classes/Core/Internal/Image+Assets.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    static var device = Image.load(using: "device")
+    static var application = Image.load(using: "application")
+    static var tools = Image.load(using: "tools")
+    static var preferences = Image.load(using: "preferences")
+    static var performance = Image.load(using: "performance")
+}
+
+extension Image {
+
+    static func load(using name: String) -> Image {
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/Internal/Image+Assets.swift:25]: I would rather make the return type as `Optional` and create additional method that will contain `defaultImage` parameter (which can be what you are returning here - it can also be defined as a default parameter) and that would resolve potential unexpected behaviour.

P.S. Also add comments ono both methods.

File: Sentinel/Classes/Core/PerformanceInfo/Items/PerformanceToolView.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import SwiftUI
+
+struct PerformanceToolView: View {
+
+    @ObservedObject var viewModel: PerformanceInfoViewModel
+
+    var body: some View {
+        HStack(spacing: 10) {
+            Text(viewModel.item.title)
+                .font(.system(size: 13, weight: .bold))
```
[the remark this answers is not in the document]
Comment (nikolamajcen) [reply Sentinel/Classes/Core/PerformanceInfo/Items/PerformanceToolView.swift:17]: I would recommend creating a new ticket which will cover this. Once you create it, you can resolve this one.

File: Sentinel/Classes/Core/ToolTableSection.swift [hunk 6]
```diff
@@ -20,11 +19,18 @@ public class ToolTableSection: NSObject {
...
-        super.init()
+    }
+}
+
+// MARK: - Identifiable conformance
+
+extension ToolTableSection: Identifiable {
+
+    public var id: String {
+        title ?? UUID().uuidString
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/ToolTableSection.swift:34]: Due to the nature of computed property and if the `title` is `nil`, the returned string will be different every time the `id` is called/used.

**Comment by nikolamajcen** [comment]
Additionally, check the Bitrise and fix tests issue - in short, change the current minimum version from iOS 18 to iOS 14.0 on the `Sentinel_Tests` target.

Everything else looks really nice! Great work 👏

File: Sentinel/Classes/Core/PerformanceInfo/Items/PerformanceToolView.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (ZvonimirMedak) [reply Sentinel/Classes/Core/PerformanceInfo/Items/PerformanceToolView.swift:17]: Created a ticket on Productive (https://app.productive.io/1-infinum/projects/1116/tasks/task/10469176)

[the remark this answers is not in the document]
Comment (nikolamajcen) [reply Sentinel/Classes/Core/PerformanceInfo/Items/PerformanceToolView.swift:17]: Perfect, this one is resolved 👍

File: Sentinel/Classes/Core/Internal/Image+Assets.swift (hunk 3 above)
Comment (ZvonimirMedak) [reply Sentinel/Classes/Core/Internal/Image+Assets.swift:10]: I created an issue here so we don't pollute Productive with issues. I think that we have a better reach here with the whole community https://github.com/infinum/ios-sentinel/issues/45

File: Sentinel/Classes/Core/ToolTableSection.swift (hunk 6 above)
Comment (ZvonimirMedak) [reply Sentinel/Classes/Core/ToolTableSection.swift:34]: Nice catch 👌 
I added it as a property which will be instantiated through the init

**Review by nikolamajcen — APPROVED** [review]
Ready to go 🚀
